model/minsky.cc

```diff
@@ -713,6 +727,101 @@ namespace minsky
        });
     canvas.requestRedraw();
   }
+  
+  /*
+  For using GSL non-linear least square fitting routines
+  patterned on first Large Nonlinear Least Squares Example.
+  See online documentation at https://www.gnu.org/software/gsl/doc/html/nls.html
+  */
+      
+namespace
+{	
+
+   int residualFunc(const gsl_vector * x, void *params,
+          gsl_vector * f)
+   {
+	if (params==NULL) return GSL_EBADFUNC;
+	try { 	  
+		
+	      auto pars=static_cast<OptParams*>(params);  
+	      
+          int err = GSL_SUCCESS;
+          auto n = ValueVector::stockVars.size();
+          vector<double> xx(n);
+          
+          memcpy(xx.data(),x,n);
```

Comment:
The correct way of doing this should be
``vector<double> xx(x->size); 
memcpy(x.data(), x->data, x->size * sizeof(double));``

Doing it the way your doing will most likely lead to memory corruption, I don't know why it is working for your.
See [gsl_vector](https://www.gnu.org/software/gsl/doc/html/vectors.html)

Reply:
Note - the above code is only valid if x->stride == 1.

Reply:
If stride>1, then you may be able to use gsl_vector_memcpy instead.

Reply:
Thank you, this is an important lesson. I should never assume that because gsl_vector and vector<double> are both vectors that their internal data structures will be similar.